# A quick overview of sequencing technologies

There are multiple sequencing technologies currently present on the market. These include can be broadly divided in two categories:

- Amplification-based
  - [Sanger](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC431765/pdf/pnas00043-0271.pdf) (yes, it is still around)
  - [454](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1464427/) (dead)
  - [IonTorrent](https://www.nature.com/articles/nature10242) (kinda alive)
  - [SOLiD](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2752135/) (dead)
  - [Complete genomics](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=19892942) (reincarnated by BGI)
  - [Illumina](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2581791/pdf/nihms72488.pdf) (mature but nearing midlife crisis)
  - [Element](https://www.nature.com/articles/s41587-023-01750-7) (a new superactive kid on the block)
- Single-molecule
  - [Helicos](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-s1-i14) (dead)
  - [PacBio](https://science.sciencemag.org/content/323/5910/133/) (mature)
  - [ONT](https://pubs.acs.org/doi/full/10.1021/ja1087612) (maturing teenager)

![image](https://github.com/nekrut/BMMB582/assets/4291636/95eae4bc-37ea-417f-a106-f1a6140ad1cb)
 
Today's most relevant platforms are:
  1. Illumina
  2. Element
  3. PacBio
  4. ONT

They are discusssed in this lecture.

## A bit of history

### Sanger dideoxy method

Fred Sanger took advantage of inhibitors that stop the extension of a DNA strand at particular nucleotides. These inhibitors are dideoxy analogs of normal nucleotide triphosphates:

-----

![](https://i.imgur.com/VUa996S.png)

**Sanger ddNTP gel**. From [Sanger:1977](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC431765/pdf/pnas00043-0271.pdf)

----

## Original approaches were laborious

In the original [Sanger paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC431765/pdf/pnas00043-0271.pdf) the authors sequenced bacteriophage phiX174 by using its own restriction fragments as primers. This was an ideal set up to show the proof of principle for the new method. This is because phiX174 DNA is homogeneous and can be isolated in large quantities. Now suppose that you would like to sequence a larger genome (say _E. coli_). Remember that the original version of Sanger method can only sequence fragments up to 200 nucleotides at a time. So to sequence the entire _E. coli_ genome (which by-the-way was not sequenced until [1997](http://science.sciencemag.org/content/277/5331/1453)) you would need to split the genome into multiple pieces and sequence each of them individually. This is hard, because to produce a readable Sanger sequencing gel each sequence must be amplified to a suitable amount (around 1 nanogram) and be homogeneous (you cannot mix multiple DNA fragments in a single reaction as it will be impossible to interpret the gel). Molecular cloning enabled by the availability of commercially available restriction enzymes and cloning vectors simplified this process. Until the onset of next generation sequencing in 2005 the process for sequencing looked something like this:

* (**1**) - Generate a collection of fragments you want to sequence. It can be a collection of fragments from a genome that was mechanically sheared or just a single fragment generated by PCR.
* (**2**) - These fragment(s) are then cloned into a plasmid vector (we will talk about other types of vectors such as BACs later in the course).
* (**3**) - Vectors are transformed into bacterial cells and positive colonies (containing vectors with insert) are picked from an agar plate.
* (**4**) - Each colony now represents a unique piece of DNA. 
* (**5**) - An individual colony is used to seed a bacterial culture that is grown overnight.
* (**6**) - Plasmid DNA is isolated from this culture and now can be used for sequencing because it is (1) homogeneous and (2) we now a sufficient amount.
* (**7**) - It is sequenced using universal primers. For example the image below shows a map for pGEM-3Z plasmid (a pUC18 derivative). Its multiple cloning site is enlarged and sites for **T7** and **SP6** sequencing primers are shown. These are the **pads** I'm referring to in the lecture. These provide universal sites that can be used to sequence any insert in between. 

-----

![](https://i.imgur.com/bkfrwVJ.png)

**pGEM-3Z**. Figure from Promega, Inc.

-----

Until the invention of NGS the above protocol was followed with some degree of automation. But you can see that it was quite laborious if the large number of fragments needed to be sequenced. This is because each of them needed to be subcloned and handled separately. This is in part why Human Genome Project, a subject of our next lecture, took so much time to complete. 

### Evolution of sequencing machines

The simplest possible sequencing machine is a [gel rig with polyacrylamide gel](https://en.wikipedia.org/wiki/Polyacrylamide_gel_electrophoresis). Sanger used it is his protocol obtaining the following results:

-----

![](https://i.imgur.com/7mRCGEi.png)

Figure from [Sanger et al. 1977](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC431765/pdf/pnas00043-0271.pdf).

----

Here for sequencing each fragment four separate reactions are performed (with ddA, ddT, ggC, and ddG) and four lanes on the gel are used. [One simplification of this process](http://www.jstor.org/stable/pdf/2879539.pdf) that came in the 90s was to use fluorescently labeled dideoxy nucleotides. This is easier because everything can be performed in a single tube and uses a single lane on a gel:

-----

![](https://i.imgur.com/0eC2lLT.png)

Figure from Applied Biosystems [support site](https://www3.appliedbiosystems.com/cms/groups/mcb_support/documents/generaldocuments/cms_041003.pdf).

------

However, there is still substantial labor involved in pouring the gels, loading them, running machines, and cleaning everything post-run. A significant improvement was offered by the development of capillary electrophoresis allowing automation of liquid handling and sample loading. Although several manufacturers have been developing and selling such machines a _de facto_ standard in this area was (and still is) the Applied Biosystems (ABI) Genetics and DNA Anlayzer systems. The highest throughput ABI system, 3730_xl_, had 96 capillaries and could automatically process 384 samples. 

### NGS!

384 samples may sound like a lot, but it is nothing if we are sequencing an entire genome. The beauty of NGS is that these technologies are not bound by sample handling logistics. They still require preparation of libraries, but once a library is made (which can be automated) it is processed more or less automatically to generate multiple copies of each fragment (in the case of 454, Illumina, and Ion Torrent) and loaded onto the machine, where millions of individual fragments are sequenced simultaneously. The following videos and slides explains how these technologies work.

### Historic NGS technologies: 454

454 Technology is a massively parallel modification of [pyrosequencing](http://genome.cshlp.org/content/11/1/3) technology. Incorporation of nucleotides are registered by a [CCD](https://en.wikipedia.org/wiki/Charge-coupled_device) camera as a flash of light generated from the interaction between ATP and Luciferin. The massive scale of 454 process is enabled by generation of a population of beads carrying multiple copies of the same DNA fragment. The beads are distributed across a microtiter plate where each well of the plate holding just one bead. Thus every unique coordinate (a well) on the plate generates flashes when a nucleotide incorporation event takes plate. This is "monochrome" technique: flash = nucleotide is incorporated; lack of flash = no incorporation. Thus to distinguish between A, C, G, and T individual nucleotides are "washed" across the microtiter plate at discrete times: if **A** is being washed across the plate and a flash of light is emitted, this implies that A is present in the fragment being sequenced. 

454 can generated fragments up 1,000 bases in length. Its biggest weakness is inability to precisely determine the length of [homopolymer runs](https://www.broadinstitute.org/crd/wiki/index.php/Homopolymer). Thus the main type if sequencing error generated by 454 are insertions and deletions (indels).

Here is a [set of slides](https://drive.google.com/file/d/1V9XwJo2NdmNclB6WCbBRzVwjm1WPxPcG/view?usp=sharing) for a more detailed look.

## Illumina

[Illumina](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2581791/pdf/nihms72488.pdf) (originally Solexa) is the second highly parallel sequencing technology. 
It is by far the most widely used technology of the day:

### Pros

- Reads up to 300 bp in single or paired end configuration
- Straightforward multiplexing
- Relatively low error rate
- Very high throughput

### Cons

- Reads up to 300 bp
- Relatively high cost per base for smaller machines

### How does it work

Let's look at these [slides](https://docs.google.com/presentation/d/1oF0dqWpGZOytQeuhgoAK0m53-xmGRnqGkJaYEasuSNo/edit?usp=sharing)

## Element Biosciences

> [!NOTE]
> Figures are from [Arslan et al. 2023](https://www.nature.com/articles/s41587-023-01750-7). Clicking on each figure will bring you to the original web page.

Element is similar to Illuimina in that it uses amplified clusters of DNA fragments within a flowcell. However, it uses a very different approach to detection and extension. Nucleotide detection is perfomed by using polymerase for incorporation of nucleotides attached to a dye-labelled polymer. These nucleotides are NOT covalently linked with the nascent DNA strand. Instead they are washed out and then filled with an unlabelled 3'-blocked nucleotide. Block is removed and cycle is repeated:

[![image](https://github.com/nekrut/BMMB582/assets/4291636/fcd30988-1a88-4325-8958-56c8968371fb)](https://www.nature.com/articles/s41587-023-01750-7/figures/1)

This changes binding/dissociation kinetics (in Element an "avidite" with multiple nucleotides is bound to template, while in Illumina a single nucleotides is used) and significantly inproves signal-to-noise ratio:

[![image](https://github.com/nekrut/BMMB582/assets/4291636/98affedf-fb88-48a9-9f5b-711e36ba2015)](https://www.nature.com/articles/s41587-023-01750-7/figures/3)

It is also very good in dealing with homopolymer runs:

[![image](https://github.com/nekrut/BMMB582/assets/4291636/98ee4b48-4b40-41cd-b793-b61bb2c74480)](https://www.nature.com/articles/s41587-023-01750-7/figures/5)

The data generated by ElemenBio machines is very similar with Illumina data - it just has higher quality values and lower cost per base for low to mid-scale applivcations. 

### Oxford Nanopore

> [!NOTE]
> This text and figures are based on the offician Oxford nanopоре documentation from the official [nanoporetech](https://community.nanoporetech.com/) site and official [videos](https://www.youtube.com/@OxfordNanoporeTechnologies).

The basic principle of the Oxford Nanopore Technology (ONT) is straightforward. Molecules are driven through a pore in a membrane. As a molecule moves through membrane it blocks current. In fact the extent of "blockage" is sequence specific generating using current change profiles that can be "translated" into sequence.    

![](https://i.imgur.com/e6zA8O2.png)

### Length-specific translocation speed

It began as technique for measuring polynecleotide length

[![](https://i.imgur.com/xl8KQjC.png)](https://www.pnas.org/doi/full/10.1073/pnas.93.24.13770)

### Blockage can be used to identify nucleotides

[![](https://i.imgur.com/1dzTNa8.png)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2683137/)

![](https://i.imgur.com/Do6TLIh.png)


## Speed can be controlled enzymatically

followed by experimentation on controlling the translocation speed using a variety of enzymes including a DNA polymerase:

[![](https://i.imgur.com/qipZfI0.png)](https://pubs.acs.org/doi/full/10.1021/ja1087612)


![](https://i.imgur.com/Hrnls6l.png)

### ONT also requires libraries

![](https://i.imgur.com/CeecGrO.png)

### Duplex versus simplex

![](https://i.imgur.com/lW90eiu.png)

![](https://i.imgur.com/R5BU1v1.png)

### Inny versus Outy

![](https://i.imgur.com/Vwh99tc.png)


![](https://i.imgur.com/6ewx95i.png)

![](https://i.imgur.com/U3ELctd.png)

![](https://i.imgur.com/HvaD9qH.png)

### ReadUntil

[ReadUntil](https://github.com/nanoporetech/read_until_api) allows "adaptive sampling" of reads in real time. In particular it makes things like [UNCALLED](https://github.com/skovaka/UNCALLED) possible:

[![](https://i.imgur.com/GXs57S3.png)](https://www.nature.com/articles/s41587-020-0731-9)

The general idea behind UNCALLED is summarized in this figure from the above manuscript:

-----

![](https://i.imgur.com/5itIs1D.png)

**a**, Overview of the algorithm: inputs are an FM index built from the DNA reference and the raw nanopore signal. The signal is converted to events, and the log probability of events matching each k-mer is computed. All paths through the FM index that are consistent with k-mers matching each event above a threshold are searched, conceptually forming a forest of trees (Extended Data Fig. 1 provides more details). **b**, Boxplots showing the speed of UNCALLED mapping of 100,000 E. coli reads to the *E. coli* K12 reference genome (kb s–1, left) and total number of milliseconds required to map reads (right). Center lines represent the median, box limits represent the upper and lower quartiles and whiskers represent the 5 and 95% confidence intervals. **c**, Percentage of mapped reads that can be confidently placed within a certain number of base pairs of sequencing. Note the ONT MinION sequences at approximately 450 bp s–1. **b,c**, Only reads that were mapped by UNCALLED are considered.


### [Software flow](https://github.com/nanoporetech)

[POD5](https://github.com/nanoporetech/pod5-file-format) -> [Dorado](https://github.com/nanoporetech/dorado)/[Remora](https://github.com/nanoporetech/remora)

Open [this](https://colab.research.google.com/drive/1GdUhZMmopyUMNTMEJh0TE9GtoIiO-U0-?usp=sharing) notebook to have a look at raw data.

## PacBio

PacBio was the second (after [Helicos](https://en.wikipedia.org/wiki/Helicos_single_molecule_fluorescent_sequencing)) single molecule sequencing (SMS) technology on the market.  Today it is an important technology enabling assembly of complex genomes and transcriptomes. 

### Fundamentals

The era of Pacific Biosciences begins with a publication by [Eid et al. 2009](https://science.sciencemag.org/content/323/5910/133/). Yet as was the case with other technologies it did not appear out of thin air. One of the key publications (written by some of the same authors) predating the birth of PacBio is the one by [Levene et al. 2003](http://dx.doi.org/10.1126/science.1079700). In particular it had the following figure:


-----

![](https://i.imgur.com/PmJjCUr.png)

**Figure 1** from [Levene et al. 2003](http://dx.doi.org/10.1126/science.1079700). An apparatus for single-molecule enzymology using zero-mode waveguides.

----

The overall idea of this device is that it can detect fluorescent ligands (such as, for example, labeled dNTPs) in a very small volume: 10<sup>-21</sup> litre. Specifically, envision a glass slide fused to a metal (Aluminum) film with tiny holes. In fact, the diameter of the holes is smaller than the wavelength of light that is used to illuminate the slide. As a result only molecules at the bottom of this "hole" will be detectable. This allows to track single molecule dynamics. Now imagine that you put a single DNA polymerase molecule at the bottom of such a "hole". The polymerase will be pulling dNTPs close to the bottom of the "hole" at it performs polymerization reaction. Thus, only nucleotides that are being added to DNA strand will be detected by the device at any given time. It now make a movie of this process you are recording real time polymerization kinetics. And this is exactly what PacBio process does:

----
![](https://i.imgur.com/TzQWgr7.png)

**Figure 1** from [Eid et al. 2009](https://science.sciencemag.org/content/323/5910/133/)

----

Because the "movie" recorded by the machine contains temporal component of the process (how long does it take for each base to be incorporated to the nascent chain) this information can be directly used to identify modified bases in the template as was shown by [Flushberg et al. 2010](http://www.nature.com/doifinder/10.1038/nmeth.1459):

----

![](https://i.imgur.com/LG6Zkmi.png)

**Figure 1** from [Flushberg et al. 2010](http://www.nature.com/doifinder/10.1038/nmeth.1459).

----

### Real life applications and accuracy

One of the major drawbacks of SNS technologies is a relatively high error. In the case of PacBio it is somewhere between 10 to 25%. However, because PacBio uses library produced by ligating bell-shaped adapters to DNA molecules, a single circular molecule can be sequenced multiple times allowing for error correction:

----

![](https://i.imgur.com/NEvb7Rp.png)

**Figure 1a** from [Wenger et al. 2019](http://dx.doi.org/10.1038/s41587-019-0217-9).

----

Performing multiple passes allows for high accuracy:

-----

![](https://i.imgur.com/BVM5c1g.png)

**Figure 2b** from [Wenger et al. 2019](http://dx.doi.org/10.1038/s41587-019-0217-9).

----

with reads of approximately 13kb:

-----

![](https://i.imgur.com/nhQAQ1y.png)

**Figure 1c** from [Wenger et al. 2019](http://dx.doi.org/10.1038/s41587-019-0217-9).

> [!IMPORTANT]
> Note that it is also possible to read a longer insert just  generating what is called Continuous Long Reads (**CLRs**). These are obviously much longer but are less accurate. Thus current PacBio systems produce two types of reads: Circular Consensus Reads (**CCR**) and Continuous Long Reads (CLRs). A subset of high quality CCR reads (with base $Q > 20$) is called **HiFi** reads.

### Practicalities

In addition to sequencing data and base qualities PacBio machines produce kinetics data. In particular:

- Intra Pulse Duration values (IDPs)
- Pulse width values

Older machines were reporting all this info as [Hierarchical Data Format (HDF)](https://en.wikipedia.org/wiki/Hierarchical_Data_Format). Newer machines are using a special subset of [BAM](https://pacbiofileformats.readthedocs.io/en/12.0/BAM.html) format to report this information. Because of all these additional data PacBio datasets tend to be quite large:

![](https://i.imgur.com/nPJccSr.png)

You can look at a small fragment of the `*.hifi_reads.bam` file in its textual (SAM) representation [here](https://github.com/nekrut/BMMB554/blob/master/2021/data/sample.hifi_reads.sam).



