# A quick overview of sequencing technologies

There are multiple sequencing technologies currently present on the market. These include can be broadly divided in two categories:

- Amplification-based
  - [Sanger](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC431765/pdf/pnas00043-0271.pdf) (yes it is still around)
  - [454](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1464427/) (dead)
  - [IonTorrent](https://www.nature.com/articles/nature10242) (kinda alive)
  - [SOLiD](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2752135/) (dead)
  - [Complete genomics](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Abstract&list_uids=19892942) (reincarnated by BGI)
  - [Illumina](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2581791/pdf/nihms72488.pdf) (mature but nearing midlife crisis)
  - [Element](https://www.nature.com/articles/s41587-023-01750-7) (a new superactive kid on the block)
- Single-molecule
  - [Helicos](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-s1-i14) (dead)
  - [PacBio](https://science.sciencemag.org/content/323/5910/133/) (mature)
  - [ONT](https://pubs.acs.org/doi/full/10.1021/ja1087612) (maturing teenager)

![image](https://github.com/nekrut/BMMB582/assets/4291636/95eae4bc-37ea-417f-a106-f1a6140ad1cb)
 
Today's most relevant platforms are:
  1. Illumina
  2. Element
  3. PacBio
  4. ONT

They are discusssed in this lecture.

## Illumina

[Illumina](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2581791/pdf/nihms72488.pdf) (originally Solexa) is the second highly parallel sequencing technology. 
It is by far the most widely used technology of the day:

### Pros

- Reads up to 300 bp in single or paired end configuration
- Straightforward multiplexing
- Relatively low error rate
- Very high throughput

### Cons

- Reads up to 300 bp
- Relatively high cost per base for smaller machines

### How does it work

Let's look at these [slides](https://docs.google.com/presentation/d/1oF0dqWpGZOytQeuhgoAK0m53-xmGRnqGkJaYEasuSNo/edit?usp=sharing)

## Element Biosciences

> [!NOTE]
> Figures are from [Arslan et al. 2023](https://www.nature.com/articles/s41587-023-01750-7). Clicking on each figure will bring you to the original web page.

Element is similar to Illuimina in that it uses amplified clusters of DNA fragments within a flowcell. However, it uses a very different approach to detection and extension. Nucleotide detection is perfomed by using polymerase for incorporation of nucleotides attached to a dye-labelled polymer. These nucleotides are NOT covalently linked with the nascent DNA strand. Instead they are washed out and then filled with an unlabelled 3'-blocked nucleotide. Block is removed and cycle is repeated:

[![image](https://github.com/nekrut/BMMB582/assets/4291636/fcd30988-1a88-4325-8958-56c8968371fb)](https://www.nature.com/articles/s41587-023-01750-7/figures/1)

This changes binding/dissociation kinetics (in Element an "avidite" with multiple nucleotides is bound to template, while in Illumina a single nucleotides is used) and significantly inproves signal-to-noise ratio:

[![image](https://github.com/nekrut/BMMB582/assets/4291636/98affedf-fb88-48a9-9f5b-711e36ba2015)](https://www.nature.com/articles/s41587-023-01750-7/figures/3)

It is also very good in dealing with homopolymer runs:

[![image](https://github.com/nekrut/BMMB582/assets/4291636/98ee4b48-4b40-41cd-b793-b61bb2c74480)](https://www.nature.com/articles/s41587-023-01750-7/figures/5)

The data generated by ElemenBio machines is very similar with Illumina data - it just has higher quality values and lower cost per base for low to mid-scale applivcations. 

## Oxford Nanopore

> [!NOTE]
> This text and figures are based on the offician Oxford nanopоре documentation from the official [nanoporetech](https://community.nanoporetech.com/) site and official [videos](https://www.youtube.com/@OxfordNanoporeTechnologies).

The basic principle of the Oxford Nanopore Technology (ONT) is straightforward. Molecules are driven through a pore in a membrane. As a molecule moves through membrane it blocks current. In fact the extent of "blockage" is sequence specific generating using current change profiles that can be "translated" into sequence.    

![](https://i.imgur.com/e6zA8O2.png)

### Length-specific translocation speed

It began as technique for measuring polynecleotide length

[![](https://i.imgur.com/xl8KQjC.png)](https://www.pnas.org/doi/full/10.1073/pnas.93.24.13770)

## Blockage can be used to identify nucleotides

[![](https://i.imgur.com/1dzTNa8.png)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2683137/)

![](https://i.imgur.com/Do6TLIh.png)


## Speed can be controlled enzymatically

followed by experimentation on controlling the translocation speed using a variety of enzymes including a DNA polymerase:

[![](https://i.imgur.com/qipZfI0.png)](https://pubs.acs.org/doi/full/10.1021/ja1087612)


![](https://i.imgur.com/Hrnls6l.png)

## ONT also requires libraries

![](https://i.imgur.com/CeecGrO.png)

## Duplex versus simplex

![](https://i.imgur.com/lW90eiu.png)

![](https://i.imgur.com/R5BU1v1.png)

## Inny versus Outy

![](https://i.imgur.com/Vwh99tc.png)


![](https://i.imgur.com/6ewx95i.png)

![](https://i.imgur.com/U3ELctd.png)

![](https://i.imgur.com/HvaD9qH.png)

## ReadUntil

[ReadUntil](https://github.com/nanoporetech/read_until_api) allows "adaptive sampling" of reads in real time. In particular it makes things like [UNCALLED](https://github.com/skovaka/UNCALLED) possible:

[![](https://i.imgur.com/GXs57S3.png)](https://www.nature.com/articles/s41587-020-0731-9)

The general idea behind UNCALLED is summarized in this figure from the above manuscript:

-----

![](https://i.imgur.com/5itIs1D.png)

**a**, Overview of the algorithm: inputs are an FM index built from the DNA reference and the raw nanopore signal. The signal is converted to events, and the log probability of events matching each k-mer is computed. All paths through the FM index that are consistent with k-mers matching each event above a threshold are searched, conceptually forming a forest of trees (Extended Data Fig. 1 provides more details). **b**, Boxplots showing the speed of UNCALLED mapping of 100,000 E. coli reads to the *E. coli* K12 reference genome (kb s–1, left) and total number of milliseconds required to map reads (right). Center lines represent the median, box limits represent the upper and lower quartiles and whiskers represent the 5 and 95% confidence intervals. **c**, Percentage of mapped reads that can be confidently placed within a certain number of base pairs of sequencing. Note the ONT MinION sequences at approximately 450 bp s–1. **b,c**, Only reads that were mapped by UNCALLED are considered.


## [Software flow](https://github.com/nanoporetech)

[POD5](https://github.com/nanoporetech/pod5-file-format) -> [Dorado](https://github.com/nanoporetech/dorado)/[Remora](https://github.com/nanoporetech/remora)

Open [this](https://colab.research.google.com/drive/1GdUhZMmopyUMNTMEJh0TE9GtoIiO-U0-?usp=sharing) notebook to have a look at raw data.
